src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java

```diff
@@ -403,4 +404,93 @@ public void shouldWorkWithNullValues(boolean consistentSnapshot, boolean useSnap
             assertValueField(records.get(i), "after/hours/value", null);
         }
     }
+
+    @ParameterizedTest
+    @MethodSource("io.debezium.connector.yugabytedb.TestHelper#streamTypeProviderForStreaming")
+    public void shouldResumeLargeTransactions(boolean consistentSnapshot, boolean useSnapshot) throws Exception {
+        /*
+         * The goal behind this test is to make sure that when explicit checkpoints are not being
+         * updated, the safe time should also not be moving forward since it will cause data loss
+         * by filtering the records.
+         *
+         * 1. Start connector
+         * 2. Stop updating the explicit checkpoint
+         * 3. Execute transactions - since the checkpoints are not being updated, nothing will go
+         *    to the cdc_state table
+         * 4. Restart the connector - once the connector restarts, it should consume all the events
+         *    from the starting without any data loss
+         */
+        TestHelper.dropAllSchemas();
+        TestHelper.executeDDL("yugabyte_create_tables.ddl");
+
+        YugabyteDBStreamingChangeEventSource.TRACK_EXPLICIT_CHECKPOINTS = true;
+
+        String dbStreamId = TestHelper.getNewDbStreamId(DEFAULT_DB_NAME, "t1", consistentSnapshot, useSnapshot);
+        Configuration.Builder configBuilder = TestHelper.getConfigBuilder("public.t1", dbStreamId);
+        startEngine(configBuilder);
+        awaitUntilConnectorIsReady();
+
+        YBClient ybClient = TestHelper.getYbClient(TestHelper.getMasterAddress());
+        YBTable ybTable = TestHelper.getYbTable(ybClient, "t1");
+        assertNotNull(ybTable);
+
+        Set<String> tablets = ybClient.getTabletUUIDs(ybTable);
+
+        String tabletId = tablets.iterator().next();
+        assertNotNull(tabletId);
+        assertEquals(1, tablets.size());
+
+        // This check will verify that we are getting some explicit checkpoint values, once that is confirmed, stop updating the explicit checkpoints.
+        Awaitility.await()
+          .atMost(Duration.ofMinutes(1))
+          .until(() -> YugabyteDBStreamingChangeEventSource.TEST_explicitCheckpoints.get(tabletId) != null);
+        YugabyteDBStreamingChangeEventSource.UPDATE_EXPLICIT_CHECKPOINT = false;
```

Comment:
How do we get explicit checkpoint without inserting anything?

Reply:
If the response is empty i.e. has no records even then we keep moving the checkpoint forward with the response. Now the initial `GetChanges` calls give us a DDL and then the response keeps moving forward - the safe time value keeps moving forward, consequently the explicit checkpoint also has a value.

Code reference:
https://github.com/yugabyte/debezium-connector-yugabytedb/blob/main/src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java#L762